src/dpp/guild.cpp

```diff
@@ -211,19 +232,23 @@ std::string guild_member::build_json(bool with_id) const {
 		}
 	}
 	
-	if (!this->nickname.empty()) {
-		j["nick"] = this->nickname;
-	} else {
-		j["nick"] = json::value_t::null;
+	if (this->flags & gm_nickname_action) {
+		if (!this->nickname.empty()) {
+			j["nick"] = this->nickname;
+		} else {
+			j["nick"] = json::value_t::null;
+		}
 	}
 
-	if (!this->roles.empty()) {
-		j["roles"] = {};
-		for (auto & role : this->roles) {
-			j["roles"].push_back(std::to_string(role));
+	if (this->flags & gm_roles_action) {
+		if (!this->roles.empty()) {
```

Comment:
This whole branch is redundant, the `for` loop will be a no-op anyways if it's empty

Reply:
Huh
I am talking about this->roles yes
You check if it's empty then you loop over it to push_back roles in it, the loop is also a check that it's empty